compiler/src/typed/dependency_graph.re

```diff
@@ -0,0 +1,297 @@
+/**
+ Data Structure for managing graph of module dependencies during compilation
+ */
+open Graph;
+
+module type Dependency_value = {
+  type t;
+  let get_dependencies: t => list(t);
+  let get_filename: t => string;
+  let is_up_to_date: t => bool;
+  let check_up_to_date: t => unit;
+  let compile_module: (~loc:Grain_parsing.Location.t=?, t) => unit;
+  let compare: t => t => int;
+  let hash: t => int;
+  let equal: t => t => bool;
+};
+
+
+type dependency_state =
+  | Needs_processing
+  | In_progress
+  | Processed;
+
+module type G = {
+  type t;
+  module V: Sig.COMPARABLE;
+  let iter_vertex: (V.t => unit, t) => unit;
+  let iter_succ: (V.t => unit, t, V.t) => unit;
+};
+
+module Topo_make_stable = (G: {
```

Comment:
Ignore this module (it's a copy of `Topological.Make_stable` that I was using for debugging); I'll delete it before marking ready for merge